lib/redmine_xlsx_format_issue_exporter/xlsx_export_helper.rb

```diff
@@ -110,6 +111,11 @@ def write_item(worksheet, value, row_index, column_index, cell_format, is_id_col
         return
       end
 
+      if value.class.name == 'String' && value[/-?\d+#{Regexp.escape(l(:general_csv_decimal_separator))}\d{2}/] == value
```

Comment:
Does check this line that type of value is Float?
I think using value.is_a?(Float) is better solution.
See **fetch_row_values** from [Redmine source](https://www.redmine.org/projects/redmine/repository/entry/trunk/lib/redmine/export/pdf/issues_pdf_helper.rb).

Reply:
The problem here is that the value is parsed by the csv generation component in vanilla Redmine (code below) which always returns a string with two decimal places separated by a localized separator - so the quick (and dirty) solution is to reverse the process.

`def csv_value(column, object, value)
    case column.name
    when :attachments
      value.to_a.map {|a| a.filename}.join("\n")
    else
      format_object(value, false) do |value|
        case value.class.name
        when 'Float'
          sprintf("%.2f", value).gsub('.', l(:general_csv_decimal_separator))
        when 'IssueRelation'
          value.to_s(object)
        when 'Issue'
          if object.is_a?(TimeEntry)
            value.visible? ? "#{value.tracker} ##{value.id}: #{value.subject}" : "##{value.id}"
          else
            value.id
          end
        else
          value
        end
      end
    end
  end`